src/components/checkbox/checkbox.test.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { createRef } from 'react'
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Checkbox } from './checkbox'
+
+test('renders checkbox label correctly', () => {
+  render(
+    <Checkbox
+      label="Checkbox label"
+      onChange={(event) => {
+        event?.target.checked
+      }}
+    />,
+  )
+  expect(screen.getByText('Checkbox label')).toBeInTheDocument()
+})
+
+test('onChange event of Checkbox working correctly', () => {
+  const onChange = jest.fn((event) => event.target.checked)
+  render(<Checkbox label="Checkbox label" onChange={onChange} />)
+  fireEvent.click(screen.getByText('Checkbox label'))
+  expect(onChange).toBeCalled()
+  expect(onChange.mock.results[0].value).toBe(true)
+})
+
+test('renders checkbox icon container style correctly', () => {
+  let checked = true
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={(event) => {
+        checked = event?.target.checked
+      }}
+    />,
+  )
+  expect(
+    screen.getByText('Checkbox label').parentElement?.firstChild,
+  ).toHaveClass('bg-blue-500')
+})
+
+test('renders indeterminate state correctly', () => {
+  let checked: boolean | 'indeterminate' = 'indeterminate'
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={(event) => {
```

Comment:
Remove `onChange` when are not testing on change or firing any event. All tests should be isolated and only test a particular case.